In [180]:
import pandas as pd

# 번역할 파일 정제

In [187]:
# 번역할 파일 불러오기
papago = pd.read_csv('C:/PYTHON/G_ML/해커톤/data/카카오(kor).csv')
papago

전체날짜  종목명                                                 전체
0     2022.06.01 05:23  카카오  이시간에 KAKAO메인이 조선이 싹쓸이하네...메인 7개 뉴스중에 5개가 조선1개가...
1     2022.06.01 08:44  카카오  돈은 많이 받으면서근무시간에 오디오라인 열어놓는게 불만이라니\n그냥 집에서 누워서 ...
2     2022.06.01 08:59  카카오  유영두 할아버지 넘 실망이에요~~^^토른방에 남겨진 유영두 할아버지에 대한 댓글 입...
3     2022.06.01 09:29  카카오  빡빡이 하는짓 보면 모르나김범수가 일궈낸 자리 편하게 앉아서 메타버스 빌미로 날로 ...
4     2022.06.01 09:37  카카오  한오백년 전쟁하자는 미필 양키괴뢰!!!우리는 원래 하나였으나, 미,소에 의해 강제로...
...                ...  ...                                                ...
6615  2022.06.30 23:31  카카오  이거보고 윤석열이 왕따라고 생각하면https://youtu.be/C2-HTUyE6Y...
6616  2022.06.30 23:34  카카오                         잠이 안온다내일 인버스에 얼마나 꽂힐지..돈복사
6617  2022.06.30 23:40  카카오                               이거 액분전에 평균얼마엿어요궁금합니다
6618  2022.06.30 23:52  카카오                   나스닥 7천간다.\n\nKAKAO는 3만 간다.\n\nㅋㅋ
6619  2022.06.30 23:54  카카오            KAKAO 계열사이 쉐리들은 왜 배당을 안줌? 생각할수록 아주 괘씸하네

[6620 rows x 3 columns]

In [188]:
papago.isna().sum() # 결측치 확인

전체날짜    0
종목명     0
전체      0
dtype: int64

In [189]:
# 번역할 데이터
a = papago['전체']
a

0       이시간에 KAKAO메인이 조선이 싹쓸이하네...메인 7개 뉴스중에 5개가 조선1개가...
1       돈은 많이 받으면서근무시간에 오디오라인 열어놓는게 불만이라니\n그냥 집에서 누워서 ...
2       유영두 할아버지 넘 실망이에요~~^^토른방에 남겨진 유영두 할아버지에 대한 댓글 입...
3       빡빡이 하는짓 보면 모르나김범수가 일궈낸 자리 편하게 앉아서 메타버스 빌미로 날로 ...
4       한오백년 전쟁하자는 미필 양키괴뢰!!!우리는 원래 하나였으나, 미,소에 의해 강제로...
                              ...                        
6615    이거보고 윤석열이 왕따라고 생각하면https://youtu.be/C2-HTUyE6Y...
6616                           잠이 안온다내일 인버스에 얼마나 꽂힐지..돈복사
6617                                 이거 액분전에 평균얼마엿어요궁금합니다
6618                     나스닥 7천간다.\n\nKAKAO는 3만 간다.\n\nㅋㅋ
6619              KAKAO 계열사이 쉐리들은 왜 배당을 안줌? 생각할수록 아주 괘씸하네
Name: 전체, Length: 6620, dtype: object

# [ 파파고 번역1 ] : openAPI로 csv파일 번역

In [239]:
import os
import sys
import json
import urllib.request


url = 'https://openapi.naver.com/v1/papago/n2mt'
client_id = 'qRRDuul_fIUg_jytEypx'
client_secret = 'eMUZEK5G_1'

text = a
afterTranslate = []
# Papago API 를 사용할 때는 google 번역 API와 다르게 원본 언어와 타겟 언어를 반드시 지정을 해야합니다
source = 'ko'
target = 'en'

encText = urllib.parse.quote(str(a))
data = f'source={source}&target={target}&text=' + encText
request = urllib.request.Request(url)
request.add_header("X-Naver-Client-Id", client_id)
request.add_header("X-Naver-Client-Secret", client_secret)
response = urllib.request.urlopen(request, data=data.encode("utf-8"))
rescode = response.getcode()

if (rescode == 200):
    response_body = response.read()
    decode = json.loads(response_body.decode('utf-8'))
    # print(decode)
    result = decode['message']['result']['translatedText']
    print(result)
else:

    print("Error Code:" + rescode)


At this time, KAKAO's main dish is being swept away by Joseon.Out of the 7 main news, 5 of them are Chosun.
I get paid a lot, but I don't like to keep the audio line open during work hours.
2. I'm so disappointed^^ Comments about grandfather Yoo Young-doo left in Thor's room...
You may not know if I'm shaved, but I'll sit comfortably in the seat that Kim Bum Soo created and use it as an excuse for metaverse.
Four hundred-year-old Yankee bullies!We were originally one, but forced by Mi and So...
                              ...                        
6615 If you think Yoon Seokyeol is an outcast, https://youtu.be/C2-HTUyE6Y...
6616, I can't sleepI don't know how much I'll be hooked on the Inverse tomorrow.money copy
6617, how much was the average before the liquid level?I'm curious.
6618 NASDAQ: Seven thousand.\n\nKAO goes to 30,000.\n\nLOL
Why don't the 6619 KAKAO affiliates pay dividends? The more I think about it, the more disgusting it is
Name: overall, Length: 6620, dtype: object

In [263]:
df1 = pd.DataFrame(data = [result], columns=['1'] ,index=[0])
df1

1
0  At this time, KAKAO's main dish is being swept...

## 문제점1
> 번역기 코드를 돌린 결과, 종토방(영어ver) 글들이 하나로 합쳐진 채로 값이 나옴  

    ---> 해결방법 1: 어떤 값(예: " or ' or \n)을 기준으로 분할 -> but 명확한 기준이 되는 값이 없음

 ### - 엔터 기준으로 분할


In [249]:
aa = result.split("\n")
aa

["At this time, KAKAO's main dish is being swept away by Joseon.Out of the 7 main news, 5 of them are Chosun.",
 "I get paid a lot, but I don't like to keep the audio line open during work hours.",
 "2. I'm so disappointed^^ Comments about grandfather Yoo Young-doo left in Thor's room...",
 "You may not know if I'm shaved, but I'll sit comfortably in the seat that Kim Bum Soo created and use it as an excuse for metaverse.",
 'Four hundred-year-old Yankee bullies!We were originally one, but forced by Mi and So...',
 '                              ...                        ',
 '6615 If you think Yoon Seokyeol is an outcast, https://youtu.be/C2-HTUyE6Y...',
 "6616, I can't sleepI don't know how much I'll be hooked on the Inverse tomorrow.money copy",
 "6617, how much was the average before the liquid level?I'm curious.",
 '6618 NASDAQ: Seven thousand.\\n\\nKAO goes to 30,000.\\n\\nLOL',
 "Why don't the 6619 KAKAO affiliates pay dividends? The more I think about it, the more disgusting it

In [275]:
len(aa)

12

### - " 기준으로 분할

In [274]:
bb = result.split('"')
bb

["At this time, KAKAO's main dish is being swept away by Joseon.Out of the 7 main news, 5 of them are Chosun.\nI get paid a lot, but I don't like to keep the audio line open during work hours.\n2. I'm so disappointed^^ Comments about grandfather Yoo Young-doo left in Thor's room...\nYou may not know if I'm shaved, but I'll sit comfortably in the seat that Kim Bum Soo created and use it as an excuse for metaverse.\nFour hundred-year-old Yankee bullies!We were originally one, but forced by Mi and So...\n                              ...                        \n6615 If you think Yoon Seokyeol is an outcast, https://youtu.be/C2-HTUyE6Y...\n6616, I can't sleepI don't know how much I'll be hooked on the Inverse tomorrow.money copy\n6617, how much was the average before the liquid level?I'm curious.\n6618 NASDAQ: Seven thousand.\\n\\nKAO goes to 30,000.\\n\\nLOL\nWhy don't the 6619 KAKAO affiliates pay dividends? The more I think about it, the more disgusting it is\nName: overall, Length: 66

In [276]:
len(bb)

1

### - ' 기준으로 분할

In [277]:
cc = result.split("'")
cc

['At this time, KAKAO',
 's main dish is being swept away by Joseon.Out of the 7 main news, 5 of them are Chosun.\nI get paid a lot, but I don',
 't like to keep the audio line open during work hours.\n2. I',
 'm so disappointed^^ Comments about grandfather Yoo Young-doo left in Thor',
 's room...\nYou may not know if I',
 'm shaved, but I',
 'll sit comfortably in the seat that Kim Bum Soo created and use it as an excuse for metaverse.\nFour hundred-year-old Yankee bullies!We were originally one, but forced by Mi and So...\n                              ...                        \n6615 If you think Yoon Seokyeol is an outcast, https://youtu.be/C2-HTUyE6Y...\n6616, I can',
 't sleepI don',
 't know how much I',
 'll be hooked on the Inverse tomorrow.money copy\n6617, how much was the average before the liquid level?I',
 'm curious.\n6618 NASDAQ: Seven thousand.\\n\\nKAO goes to 30,000.\\n\\nLOL\nWhy don',
 't the 6619 KAKAO affiliates pay dividends? The more I think about it, the more

In [279]:
len(cc)

12

In [244]:
len(aa)

12

### - 엔터, ", ' 기준으로 분할

In [280]:
type(result)
_result = []
for s in result.split('.'):
    s = s.replace('\n', '').replace(',','').replace('"','').strip()
    if(len(s) != 0):
        _result.append(s)
_result

["At this time KAKAO's main dish is being swept away by Joseon",
 'Out of the 7 main news 5 of them are Chosun',
 "I get paid a lot but I don't like to keep the audio line open during work hours",
 '2',
 "I'm so disappointed^^ Comments about grandfather Yoo Young-doo left in Thor's room",
 "You may not know if I'm shaved but I'll sit comfortably in the seat that Kim Bum Soo created and use it as an excuse for metaverse",
 'Four hundred-year-old Yankee bullies!We were originally one but forced by Mi and So',
 '6615 If you think Yoon Seokyeol is an outcast https://youtu',
 'be/C2-HTUyE6Y',
 "6616 I can't sleepI don't know how much I'll be hooked on the Inverse tomorrow",
 "money copy6617 how much was the average before the liquid level?I'm curious",
 '6618 NASDAQ: Seven thousand',
 '\\n\\nKAO goes to 30000',
 "\\n\\nLOLWhy don't the 6619 KAKAO affiliates pay dividends? The more I think about it the more disgusting it isName: overall Length: 6620 dtype: object"]

In [281]:
len(_result)

14

    > 문제점1 결론: 기준을 잡고 데이터를 분할해도 6620개로 분할 안됨

# [ 파파고 번역2 ] : pypapago라이브러리 이용

In [282]:
from pypapago import Translator

# 번역할 파일명
fileName = 'C:/PYTHON/G_ML/해커톤/data/text.txt'

# 사용할 전역변수
templist = []
afterTranslate = []

# 객체 할당
translator = Translator()

# 파일 읽기모드, 줄을 읽고 \n를 제거하여 번역하기
with open(fileName, 'rt', encoding='UTF8') as f:
    lines = f.readlines()
    for line in lines:
        # '\n'엔터키''가 있을 때 교체하여 추가, 그 외 그냥 추가
        if '\n' in line:
            line = line.replace('\n', '`')
            templist.append(line)
        else:
            templist.append(line)

        # 문자열이 비어있는 리스트는 삭제
        try:
            templist.remove('')
        except:
            pass  

# 리스트 내 영어 자료들 번역하기
for i in range(len(templist)):
    result = translator.translate(templist[i], source='ko', target='en', verbose=False)
    afterTranslate.append(result)

# 초기 리스트값, 바꾼 후, 번역 후(출력테스트)
print(lines)
print(templist)
print(afterTranslate)


# 번역 후 text파일 만듦.
with open(fileName + '_translated.txt', 'w',encoding='utf8') as f:
    for i in range(len(afterTranslate)):
        data = afterTranslate[i] + '\n'
        f.write(data)
# 출처: https://ansan-survivor.tistory.com/78?category=368248 [안산드레아스:티스토리]

KeyError: 'translatedText'

## 문제점2
> KeyError: 'translatedText' 를 구글링 하고 코드를 수정해도 계속 에러 발생  

    ---> 포기

# [파파고 번역2] : openAPI로 tet파일 번역

In [113]:
import os
import sys
import urllib.request
import json
client_id = "qRRDuul_fIUg_jytEypx" 
client_secret = "eMUZEK5G_1" 
fileName = 'C:/PYTHON/G_ML/해커톤/data/text.txt'

#번역할 메모장 불러오기
with open(fileName,'r',encoding='utf8') as f:
    srcText = f.read()

encText = urllib.parse.quote(srcText)
# data = "source=ko&target=en&text=" + encText
data = "source=ko&target=en&text=%s"%urllib.parse.quote(srcText)
url = "https://openapi.naver.com/v1/papago/n2mt"
request = urllib.request.Request(url)
request.add_header("X-Naver-Client-Id",client_id)
request.add_header("X-Naver-Client-Secret",client_secret)
response = urllib.request.urlopen(request, data=data.encode("utf-8"))
rescode = response.getcode()
if(rescode==200):
    response_body = response.read()
    # print(response_body.decode('utf-8'))

    #json 형 변환
    res = json.loads(response_body.decode('utf-8'))
    from pprint import pprint
    pprint(res)

    #파일 생성
    with open('translate.txt', 'w',encoding='utf8') as f:
        f.write(res['message']['result']['translatedText'])

else:
    print("Error Code:" + rescode)

HTTPError: HTTP Error 400: Bad Request

## 문제점3
> HTTPError: HTTP Error 400: Bad Request 를 구글링 해본 결과 왜 오류가 발생했는지 답을 못찾음

    ---> 포기